tests/unit/test_study.py

```diff
@@ -0,0 +1,100 @@
+"""Tests for the Study class."""
+
+from pathlib import Path
+
+import pytest
+import pytest_mock
+
+from nipoppy.logger import get_logger
+from nipoppy.study import Study
+from tests.conftest import get_config
+
+
+@pytest.fixture
+def study(tmp_path: Path):
+    dpath_root = tmp_path / "my_study"
+    return Study(dpath_root=dpath_root)
+
+
+def test_init_default(tmp_path: Path):
+    custom_logger = get_logger(name="nipoppy.custom_logger", verbose=True)
+    study = Study(dpath_root=tmp_path, logger=custom_logger)
+    assert study.dpath_root == tmp_path
+    assert study.fpath_layout is None
+    assert study.logger == custom_logger
+    assert study.verbose is False
+
+
+@pytest.mark.parametrize("verbose", [True, False])
+def test_init_no_logger(tmp_path: Path, verbose: bool):
+    study = Study(dpath_root=tmp_path, verbose=verbose)
+    assert study.logger.name == "nipoppy.Study"
+    assert study.verbose is verbose
+
+
+@pytest.mark.parametrize("fname_layout", ["layout1.json", "layout2.json"])
+def test_layout(
+    study: Study,
+    fname_layout,
+    tmp_path: Path,
+    mocker: pytest_mock.MockFixture,
+):
+    fpath_layout = tmp_path / fname_layout
+
+    mocked_layout = mocker.patch("nipoppy.study.DatasetLayout")
+    study.fpath_layout = fpath_layout
+
+    _ = study.layout
```

Comment:
Why the `_ =` ... ? Can't we skip the assignment? Otherwise, this should be documented.